src/main/java/com/commercetools/sync/services/impl/TypeServiceImpl.java

```diff
@@ -4,27 +4,47 @@
 import com.commercetools.sync.commons.BaseSyncOptions;
 import com.commercetools.sync.commons.utils.CtpQueryUtils;
 import com.commercetools.sync.services.TypeService;
+import com.commercetools.sync.types.TypeSyncOptions;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.queries.PagedResult;
+import io.sphere.sdk.queries.QueryExecutionUtils;
 import io.sphere.sdk.types.Type;
+import io.sphere.sdk.types.TypeDraft;
+import io.sphere.sdk.types.commands.TypeCreateCommand;
+import io.sphere.sdk.types.commands.TypeUpdateCommand;
 import io.sphere.sdk.types.queries.TypeQuery;
+import io.sphere.sdk.types.queries.TypeQueryBuilder;
 
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.Consumer;
 
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toSet;
+import static org.apache.http.util.TextUtils.isBlank;
+
 /**
  * Implementation of TypeService interface.
  * TODO: USE graphQL to get only keys. GITHUB ISSUE#84
  */
 public final class TypeServiceImpl implements TypeService {
+    private static final String FETCH_FAILED = "Failed to fetch types with keys: '%s'. Reason: %s";
```

Comment:
This type service should extend the `BaseService` class which encapsulates a lot of service common logic

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/333/commits/0c449e0dd18fafee9854b59e39f78818b6972518